src/NUnitFramework/framework/Constraints/AssignableFromConstraint.cs

```diff
@@ -25,7 +26,17 @@ public AssignableFromConstraint(Type type) : base(type, "assignable from ")
         /// <returns>True if the constraint succeeds, otherwise false.</returns>
         protected override bool Matches(object? actual)
         {
-            return actual is not null && actual.GetType().IsAssignableFrom(expectedType);
+            if (expectedType is null || actual is null)
+            {
+                return false;
+            }
+
+            if (expectedType == actualType)
+            {
+                return true;
+            }
+
+            return expectedType.CanImplicitlyConvertTo(actual.GetType());
```

Comment:
I didn't realize that _actualType_ is a field set in the base class. 
This then can now be:

```suggestion
            return expectedType.CanImplicitlyConvertTo(actualType);
```